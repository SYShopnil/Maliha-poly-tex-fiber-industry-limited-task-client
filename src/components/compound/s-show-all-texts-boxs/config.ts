interface ITextDataElements {
  isChecked: boolean;
  value: number;
  elementId: string;
}

export interface ITextBoxData {
  textId: string;
  total: number;
  elements: ITextDataElements[];
}
export const demoData: ITextBoxData[] = [
  {
    textId: "#12345",
    total: 2500,
    elements: [
      {
        elementId: "123",
        isChecked: true,
        value: 500,
      },
    ],
  },
  {
    textId: "#12345",
    total: 2500,
    elements: [
      {
        elementId: "123",
        isChecked: false,
        value: 500,
      },
      {
        elementId: "1234",
        isChecked: true,
        value: 500,
      },
      {
        elementId: "1235",
        isChecked: true,
        value: 500,
      },
      {
        elementId: "1236",
        isChecked: true,
        value: 500,
      },
      {
        elementId: "1237",
        isChecked: true,
        value: 500,
      },
      {
        elementId: "1238",
        isChecked: true,
        value: 500,
      },
    ],
  },
  {
    textId: "#12345",
    total: 2500,
    elements: [
      {
        elementId: "12333",
        isChecked: true,
        value: 500,
      },
    ],
  },
];
