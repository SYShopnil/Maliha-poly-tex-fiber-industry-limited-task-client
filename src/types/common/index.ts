export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  SHeaderMain = "SHeaderMain",
  SProfile = "SProfile",
}
