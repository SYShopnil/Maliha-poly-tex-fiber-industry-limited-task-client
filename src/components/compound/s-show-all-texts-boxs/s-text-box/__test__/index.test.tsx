import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { STextBox } from "..";
import { ITextBoxData } from "../../config";

const mock_data: ITextBoxData = {
  elements: [
    {
      elementId: "123",
      isChecked: true,
      value: 250,
    },
    {
      elementId: "122",
      isChecked: true,
      value: 250,
    },
  ],
  textId: "123456",
  total: 2500,
};

describe("When Text Box Component Rendered", () => {
  it("Expect render successfully", async () => {
    render(<STextBox {...mock_data} />);
    const myElement = screen.getByRole(EDataTestId.STextBox);
    expect(myElement).toMatchSnapshot();
  });
});
