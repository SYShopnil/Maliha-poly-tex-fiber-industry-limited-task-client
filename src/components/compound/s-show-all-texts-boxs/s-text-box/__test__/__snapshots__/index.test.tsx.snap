// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Text Box Component Rendered Expect render successfully 1`] = `
<div
  class="bg-white shadow-lg rounded-lg p-6 border border-gray-200 col-span-12 md:col-span-6 lg:col-span-4"
  role="STextBox"
>
  <p
    class="text-blue-600 font-bold mb-2"
  >
    Text ID: 
    123456
  </p>
  <p
    class="text-gray-700 mb-4"
  >
    Total Amount: 
    2500
  </p>
  <div
    class="flex justify-between"
  >
    <span
      class="cursor-pointer"
      role="CButton"
    >
      <span
        class="flex justify-center items-center  "
      >
        <span
          class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EDEDED] hover:bg-[#7A67EE]  duration-[0.5s] text-[#7A67EE] hover:text-[#EDEDED]   border-[0.063rem] hover:border-[#EDEDED] border-[#7A67EE]   "
        >
          <span>
            View Details
          </span>
        </span>
      </span>
    </span>
    <span
      class="cursor-pointer"
      role="CButton"
    >
      <span
        class="flex justify-center items-center  "
      >
        <span
          class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EBF5EC] hover:bg-[#7F4D4F]  duration-[0.5s] text-[#7F4D4F] hover:text-[#EBF5EC]   border-[0.063rem] hover:border-[#EBF5EC] border-[#7F4D4F]   "
        >
          <span>
            Update
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;
